SECURITY.md

```diff
@@ -0,0 +1,31 @@
+# Security policy
+
+This document gathers security-related policies and guidelines for the codebase
+available in this repository.
+
+Authentication is a sensitive domain, and as such we designed the `@inrupt/solid-client-authn-*`
+libraries are designed with a particular attention to security. In particular, 
+we decided to apply the following rules:
+
+- Comply with the [OAuth security guidelines](https://tools.ietf.org/id/draft-ietf-oauth-security-topics-15.html)
+where applicable. This involves, among other things:
+  - No support for the implicit grant and the resource owner password grant
+  - The use of a PKCE token
+  - Binding tokens to a DPoP key to make them sender-constrained whenever possible
+- Short-lived tokens (e.g., ID token and Access token) and private cryptographic
+material (e.g. a DPoP private key) are only stored in a function closure, so that
+they may not be extracted once received from the remote server. These protected 
+elements will not be made available directly (e.g. through a function call) or
+indirectly (e.g. allowing to sign a header with the DPoP key) to any third-party.
+- Longer-lived tokens (e.g. refresh tokens) are only stored in a secure storage,
+i.e., a storage which isn't shared beyond the librarie's control. This excludes
+browser storage such as `localStorage` and `sessionStorage`.
+- If these restrictions on the high-level layer are blocking some legitimate use
+cases, the options is added when possible for advanced users to drop down to a
```

Comment:
```suggestion
Advanced users can drop down to a lower-level API where token and key management is left
```

Reply:
That's what we eventually want, but it's not currently the case, which is why I added "when possible" to give us the slack to actually implement this.

Reply:
```suggestion
A lower-level API will be added when possible where token and key management is left to the user.
```

I would maybe try formulating like the above or remove those statements altogether (until implementation catches up).
It is unclear that it's a future plan in the current formulation and I thought straight away that I'd like to check those low level APIs wondering where/what they were ^^,